packages/frontend-web/src/app/util/partial/index.ts

```diff
@@ -40,7 +40,7 @@ function makeNode<T>(): INode<T> {
 
 const mutableObjectCache = new Map<object | Array<any>, string>();
 
-function stringifyIfNecessary<T>(o: T, useEqualityForMutableObjects: boolean): T | string {
+function stringifyIfNecessary(o: any, useEqualityForMutableObjects: boolean): boolean | number | string {
```

Comment:
Would be helpful to have a comment explaining what this function does (I guess mostly what the 'necessary' in the name means)